ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/signin/clientlibs/js/signin.js

```diff
@@ -1,10 +1,15 @@
-	const findRedriectUrl = function(domId) {
+let hasCart = false;
+const findRedriectUrl = function(domId) {
 		let redirectURL = "";
 		const params = new URLSearchParams(window.location.search);
 		const keys = ['signin.html', 'signup.html'];
 		if(params.has('referer')) {
 			if(!keys.some(k => params.get('referer').includes(k) )) {
 				redirectURL = params.get('referer');
+                if(!hasCart && redirectURL.includes('checkout.html'))
+                {
+					redirectURL = domId.getAttribute("data-default");
```

Comment:
@dee: where is data-default defined in component ?

Reply:
It is already present in SignIn Form html. Like this

<form method="POST" action="" id="login_form" name="login_form" enctype="multipart/form-data"
	 class="cmp-login-form" data-default="${properties.defaultPage}.html"